fe/fe-core/src/main/java/com/starrocks/sql/optimizer/operator/ColumnFilterConverter.java

```diff
@@ -192,6 +195,31 @@ private static boolean checkDateTruncEquals(FunctionCallExpr functionCallExpr, C
                                 TIME_MAP.get(exprTimeArg) > TIME_MAP.get(callTimeArg)));
     }
 
+    // only rewrite cast invalid date value to null like cast('abc' as date)
+    private static ScalarOperator rewriteInvalidDateCast(ScalarOperator scalarOperator) {
+        ScalarOperator copy = scalarOperator.clone();
+        List<ScalarOperator> children = copy.getChildren();
+
+        for (int i = 1; i < children.size(); i++) {
```

Comment:
why does it start from 1, not 0?

Reply:
After the previous check,  it ensures the child of index 0 is the column in the predicate. We only replace other values.